Codecademy/gamut
# fix(MenuItem): fixes focus outlines and onClick support

### Overview


Adds focus-visible outlines to all variants on `MenuItem` and creates a `button` when there is an `onClick` instead of just an `li` so `onClick` will work with keyboard navigation.


### PR Checklist

- [x] Related to JIRA ticket: [WEB-2127]

#### Testing Instructions


1. Visit https://6359c2ebafcc21039a4bc4ac--gamut-preview.netlify.app//?path=/docs/brand-labs-molecules-dropdownbutton--dropdown-button
2. Navigate via keyboard to open the dropdown button menu
3. Tab through the menu items and confirm you see the focus outline for all 3 items
4. Click on each item with your mouse - the first item should redirect the page and add #, the second should console log "hi", and the third should do both
5. Navigate via keyboard and hit enter on each item - the first item should redirect the page and add #, the second should console log "hi", and the third should do both
6. Visit https://6359c2ebafcc21039a4bc4ac--gamut-preview.netlify.app//?path=/docs/molecules-menu--menu#variants
7. Navigate via keyboard through the 3 different variants and confirm you see focus visible outlines


#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/33212)  | [Monolith Env](https://pr-33212-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/463)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-463)   |




[WEB-2127]: https://codecademy.atlassian.net/browse/WEB-2127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
